src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeMetricsCollectorSpark.java

```diff
@@ -1,99 +1,256 @@
+        }
+        if (accumLevels.contains(MetricAccumulationLevel.LIBRARY)) {
+            listOfStats.addFirst(levelUp(histogramsAtRGLevel, MetricAccumulationLevel.LIBRARY));
+        }
+        if (accumLevels.contains(MetricAccumulationLevel.SAMPLE)) {
+            listOfStats.addFirst(levelUp(histogramsAtRGLevel, MetricAccumulationLevel.SAMPLE));
+        }
+        if (accumLevels.contains(MetricAccumulationLevel.ALL_READS)) {
+            listOfStats.addFirst(levelUp(histogramsAtRGLevel, MetricAccumulationLevel.ALL_READS));
+        }
 
-        collectMedianAndMAD(lengthRDD);
+        int i=0;
+        for(final SortedMap<GroupInfo, Map<SamPairUtil.PairOrientation, SortedMap<Integer, Long>>> map : listOfStats){
+            histogramsAndMetrics.add( convertSortedMapToHTSHistogram(map, histogramMADTolerance) );
+            for(final GroupInfo groupInfo : map.keySet()){ // auxiliary map
+                String name = "";
+                switch (groupInfo.level){
+                    case ALL_READS:
+                        name = ALL_READS_KEY_MACRO;
+                        break;
+                    case SAMPLE:
+                        name = groupInfo.sample;
+                        break;
+                    case LIBRARY:
+                        name = groupInfo.library;
+                        break;
+                    case READ_GROUP:
+                        name = groupInfo.readGroup;
+                        break;
+                }
+                mapFromGroupNameToIndexInList.put(name, i++);
+            }
+        }
+    }
 
-        collectSymmetricalBinWidth(lengthRDD);
+    /**
+     * Utility getter for retrieving stats info for a particular group, given its name.
+     * Stats info returned are organized by pair orientations.
+     * If a particular pair orientation is unavailable (i.e. no reads of this group has pairs of that orientation), it is not returned.
+     * @param groupName  String representation of the group's name/id, whose stats information is requested.
+     * @return           the requested group's stats information, organized by pair orientations
+     * @throws           IllegalArgumentException
+     */
+    public Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> getHistogramsAndMetrics(final String groupName) throws IllegalArgumentException {
 
-        trimHistogramAndCollectMeanAndSD(lengthRDD, HistogramMADTolerance);
+        final Integer i = mapFromGroupNameToIndexInList.get(groupName);
+        if(null==i){
+            throw new IllegalArgumentException("Requested group name doesn't exist in current data.");
+        }
+        final SortedMap<GroupInfo, Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>>> groupInfo = histogramsAndMetrics.get(i);
+        final Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> groupStats = groupInfo.values().iterator().next();
```

Comment:
I'm a little confused by the iterator().next(). I think if you do get rid of the enclosing list as you said you might above this will get simpler.